proc/sort/sort.go

```diff
@@ -61,36 +68,42 @@ func (p *Proc) run() {
 			spiller.Cleanup()
 		}
 	}()
-	var eof bool
 	var nbytes int
 	var out []zed.Value
 	for {
-		batch, err := p.parent.Pull()
+		batch, err := p.parent.Pull(false)
 		if err != nil {
-			p.sendResult(nil, err)
-			return
-		}
-		if batch == nil {
-			if eof {
+			if ok := p.sendResult(nil, err); !ok {
 				return
 			}
-			eof = true
+			continue
+		}
```

Comment:
If `err != nil`, don't we want to return regardless of what happened in `p.sendResult`?

Reply:
let's discuss.  contract is you keep going and the propagating error cancels context

Reply:
as we discussed...